packages/connext/src/Connext.ts

```diff
@@ -1,73 +1,100 @@
 
-let _chainDataCache: Map<string, ChainData> | null = null
+const ETH_TOKEN_ADDRESS = "0x0000000000000000000000000000000000000000";
 
 const _getChainData = async () => {
   if (!_chainDataCache) {
-    const chainData = await getChainData()
-    _chainDataCache = chainData
+    const chainData = await getChainData();
+    _chainDataCache = chainData;
+  }
+
+  return _chainDataCache;
+};
+
+export const getWETHAddress = async (chainId: number) => {
+  const chains = await _getChainData();
+  const domainId = chainIdToDomain(chainId);
+  const chainData = chains?.get(String(domainId));
+  const assets = Object.keys(chainData?.assetId || {});
+
+  let wethAddress;
+  for (const address of assets) {
+    if (chainData?.assetId[address].symbol === "WETH") {
+      wethAddress = address;
+      break;
+    }
   }
+  return wethAddress;
+};
 
-  return _chainDataCache
-}
-
-export const XCALL_ABI_FRAGMENTS = [
-  {
-    inputs: [
-      { internalType: 'uint32', name: '_destination', type: 'uint32' },
-      { internalType: 'address', name: '_to', type: 'address' },
-      { internalType: 'address', name: '_asset', type: 'address' },
-      { internalType: 'address', name: '_delegate', type: 'address' },
-      { internalType: 'uint256', name: '_amount', type: 'uint256' },
-      { internalType: 'uint256', name: '_slippage', type: 'uint256' },
-      { internalType: 'bytes', name: '_callData', type: 'bytes' },
-    ],
-    name: 'xcall',
-    outputs: [{ internalType: 'bytes32', name: '', type: 'bytes32' }],
-    stateMutability: 'payable',
-    type: 'function',
-  },
-  // This overloaded function is not found in the Connext ABI json for some reason
-  {
-    inputs: [
-      { internalType: 'uint32', name: '_destination', type: 'uint32' },
-      { internalType: 'address', name: '_to', type: 'address' },
-      { internalType: 'address', name: '_asset', type: 'address' },
-      { internalType: 'address', name: '_delegate', type: 'address' },
-      { internalType: 'uint256', name: '_amount', type: 'uint256' },
-      { internalType: 'uint256', name: '_slippage', type: 'uint256' },
-      { internalType: 'bytes', name: '_callData', type: 'bytes' },
-      { internalType: 'uint256', name: '_relayerFee', type: 'uint256' },
-    ],
-    name: 'xcall',
-    outputs: [{ internalType: 'bytes32', name: '', type: 'bytes32' }],
-    stateMutability: 'payable',
-    type: 'function',
-  },
-]
-
-export const bridge = async (bridge: BridgeActionParams) => {
+export const bridge = async (
+  bridge: BridgeActionParams,
+): Promise<TransactionFilter> => {
   const {
     sourceChainId,
     destinationChainId,
     contractAddress,
     tokenAddress,
     amount,
     recipient,
-  } = bridge
+  } = bridge;
+
+  const defaultContractAddress = ConnextContract[sourceChainId];
+  const destinationDomain = chainIdToDomain(destinationChainId);
+  const requiresWrapperMultisend = tokenAddress === ETH_TOKEN_ADDRESS;
```

Comment:
`ETH_TOKEN_ADDRESS` is set to the zero address, but I'm curious about why? Will the `tokenAddress` bridge param need to use zero address for Connext WETH bridging?

Reply:
Yep basically that 

https://raw.githubusercontent.com/connext/chaindata/main/crossChain.json

```
"assetId": {
      "0x0000000000000000000000000000000000000000": {
        "symbol": "ETH",
        "mainnetEquivalent": "0x0000000000000000000000000000000000000000",
        "decimals": 18,
        "coingeckoId": "ethereum"
      },
      ...
},